src/Google.Cloud.Functions.Invoker/HostingExtensions.cs

```diff
@@ -82,87 +78,128 @@ public static ILoggingBuilder AddFunctionsFrameworkConsoleLogging(this ILoggingB
             });
 
         /// <summary>
-        /// Configures the services within the given web host builder for the Functions Framework to use
+        /// Adds the services for the Functions Framework to use
         /// a target function from the given assembly. If the Functions Framework configuration (typically provided by
         /// command line arguments or environment variables) does not specify a target function, the assembly is
         /// scanned for a single compatible function type.
         /// </summary>
-        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="services">The service collection to add the service to.</param>
         /// <param name="assembly">The assembly expected to contain the Functions Framework target function.</param>
-        /// <returns>The original builder, for method chaining.</returns>
-        public static IWebHostBuilder ConfigureFunctionsFrameworkTarget(this IWebHostBuilder builder, Assembly assembly) =>
-            builder.ConfigureServices((context, serviceCollection) =>
-                HostingInternals.AddServicesForTarget(serviceCollection, HostingInternals.GetFunctionTarget(context, assembly)));
+        /// <returns>The original service collection, for method chaining.</returns>
+        public static IServiceCollection AddFunctionsFrameworkTarget(this IServiceCollection services, WebHostBuilderContext context, Assembly assembly)
```

Comment:
Not sure about this one - Add sounds like you might be able to use it multiple times, but that's not the case. Will defer for now.
In terms of using IServiceCollection, I'm not sure - it makes it a bit harder to do this elsewhere.

(Same will go for all the methods.)